crates/secrets/src/aws_secrets_manager.rs

```diff
@@ -32,12 +50,99 @@ impl AwsSecretsManager {
+    /// This function will return an error:
+    /// - If the AWS configuration cannot be loaded.
+    /// - If the call to STS `get_caller_identity` fails, which might be due to invalid or expired AWS credentials.
+    pub async fn init(self) -> Result<(), Box<dyn std::error::Error>> {
+        let config = aws_config::defaults(BehaviorVersion::v2023_11_09())
+            .load()
+            .await;
+
+        let sts_client = aws_sdk_sts::Client::new(&config);
+
+        if sts_client.get_caller_identity().send().await.is_err() {
+            return Err(Box::new(Error::InvalidAwsCredentials {}));
+        }
+
+        Ok(())
+    }
 }
 
 #[async_trait]
 impl SecretStore for AwsSecretsManager {
     #[must_use]
-    async fn get_secret(&self, _secret_name: &str) -> Option<Secret> {
+    async fn get_secret(&self, secret_name: &str) -> Option<Secret> {
+        let secret_name = format!("{SPICE_SECRET_PREFIX}{secret_name}");
+
+        tracing::trace!("Getting secret {} from AWS Secrets Manager", secret_name);
+
+        let config = aws_config::defaults(BehaviorVersion::v2023_11_09())
+            .load()
+            .await;
+
+        let asm = aws_sdk_secretsmanager::Client::new(&config);
+
+        let secret_value = match asm.get_secret_value().secret_id(&secret_name).send().await {
+            Ok(secret) => secret,
+            Err(SdkError::ServiceError(e)) => {
+                // It is expected that not all parameters are present in secrets.
+                // Warn only if it is a different error
+                if !e.err().is_resource_not_found_exception() {
+                    tracing::warn!(
+                        "Failed to get secret {} from AWS Secrets Manager: {}",
+                        secret_name,
+                        e.err()
+                    );
+                }
+                return None;
+            }
+            Err(err) => {
+                tracing::warn!(
+                    "Failed to get secret {} from AWS Secrets Manager: {}",
+                    secret_name,
+                    err
+                );
+                return None;
+            }
+        };
+
+        if let Some(secret_str) = secret_value.secret_string() {
+            match parse_json_to_hashmap(secret_str) {
+                Ok(data) => return Some(Secret::new(data)),
+                Err(err) => {
+                    tracing::warn!("Failed to parse secret {} content: {}", secret_name, err);
+                    return None;
+                }
+            }
+        }
+
         None
     }
 }
+/// Parses a JSON string into a `HashMap<String, String>`.
+///
+/// # Errors
+///
+/// This function will return an error if:
+/// - The input string cannot be parsed as a valid JSON.
+/// - The parsed JSON is not an object. This function expects the top-level JSON structure to be a JSON object (`{}`).
+pub fn parse_json_to_hashmap(
+    json_str: &str,
+) -> Result<HashMap<String, String>, Box<dyn std::error::Error>> {
```

Comment:
Snafu Error: https://github.com/spiceai/spiceai/blob/trunk/docs/dev/style_guide.md#all-errors-use-snafu-functionality

Reply:
Updated, thank you!